app/models/amazon_agent_manager.rb

```diff
@@ -0,0 +1,411 @@
+      git clone git://github.com/rbenv/ruby-build.git ~/.rbenv/plugins/ruby-build >> #{log} 2>&1
+      echo 'export PATH="$HOME/.rbenv/plugins/ruby-build/bin:$PATH"' >> ~/.bash_profile
+      source ~/.bash_profile >> #{log} 2>&1
+      echo $PATH >> #{log} 2>&1
+      rbenv install -l >> #{log} 2>&1
+      rbenv install 2.3.3 >> #{log} 2>&1
+      rbenv global 2.3.3 >> #{log} 2>&1
+      gem install bundler >> #{log} 2>&1
+      gem install rails >> #{log} 2>&1
+      gem install aws-sdk >> #{log} 2>&1
+      rbenv rehash >> #{log} 2>&1
+      ruby -v >> #{log} 2>&1
+    SETUP
+  end
+
+  def default_settings
+    <<~SETTINGS
+      echo "---" > default_ssa_config.yml
+      echo ":log_level: #{log_level}" >> default_ssa_config.yml
+      echo ":region: #{region}" >> default_ssa_config.yml
+      echo ":request_queue: #{request_queue}" >> default_ssa_config.yml
+      echo ":reply_queue: #{reply_queue}" >> default_ssa_config.yml
+      echo ":ssa_bucket: #{ssa_bucket}" >> default_ssa_config.yml
+      echo ":reply_prefix: #{reply_prefix}" >> default_ssa_config.yml
+      echo ":log_prefix: #{log_prefix}" >> default_ssa_config.yml
+      echo ":heartbeat_prefix: #{heartbeat_prefix}" >> default_ssa_config.yml
+      echo ":heartbeat_interval: #{heartbeat_interval}" >> default_ssa_config.yml
+    SETTINGS
+  end
+
+  def github_gem_file
+    <<~GEMFILE
+      echo 'source "https://rubygems.org"' > Gemfile
+      echo 'gem "manageiq-gems-pending", ">0", :require => "manageiq-gems-pending", :git => "https://github.com/ManageIQ/manageiq-gems-pending.git", :branch => "master"' >> Gemfile
+      echo 'gem "manageiq-smartstate", "~>0.1.5", :require => "manageiq-smartstate", :git => "https://github.com/ManageIQ/manageiq-smartstate.git", :branch => "master"' >> Gemfile
+      echo 'gem "amazon_ssa_support", ">0", :require => "amazon_ssa_support", :git => "https://github.com/ManageIQ/amazon_ssa_support.git", :branch => "master"' >> Gemfile
+      # Modified gems for gems-pending.  Setting sources here since they are git references
+      echo 'gem "handsoap", "~>0.2.5", :require => false, :git => "https://github.com/ManageIQ/handsoap.git", :tag => "v0.2.5-5"' >> Gemfile
+      echo 'gem "aws-sdk"' >> Gemfile
+    GEMFILE
+  end
+
+  def startup_script(log = '/var/log/miq_ssa_deploy.log')
+    <<~STARTUP
+      echo "#!/bin/sh" > start_agent.sh
+      echo "ssa_root=`bundle show amazon_ssa_support`" >> start_agent.sh
+      echo 'ssa_script="$ssa_root/tools/amazon_ssa_extract.rb"' >> start_agent.sh
+      echo 'ruby $ssa_script -l ${log_level}' >> start_agent.sh
+      chmod 755 start_agent.sh
+      echo "Agent starts ..." >> #{log} 2>&1
+      ./start_agent.sh
+    STARTUP
+  end
+
+  def rc_local
+    <<~RCLOCAL
+      echo 'source /root/.bash_profile' >> /etc/rc.d/rc.local
+      echo 'cd /opt/miq && /opt/miq/start_agent.sh &' >> /etc/rc.d/rc.local
+    RCLOCAL
+  end
+
+  def create_user_data(log = '/var/log/miq_ssa_deploy.log')
```

Comment:
Given my comment above about the script file, this method would just read the script file and encode it.
It could also set shell variables before the script contents in case you need to pass values to the script:
```ruby
userdata = <<~DATA
  VAR1=...
  VAR2=...
  VAR3=...
  #{read_script}
DATA
Base64.encode64(userdata)
```
The script could use shell functions if you want to maintain the current modular structure.